### PR TITLE
Fix `model.save` API for legacy H5 format

### PR DESCRIPTION
**Pull request by nkovela1**
`model.save` was not updated to reflect the legacy format's existence. This keeps the API in line with `keras.saving.save_model`.

### CODE REVIEW

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: keras_core/models/model.py [hunk 1]
```diff
@@ -3,10 +3,13 @@
 import os
 import warnings
 
+from absl import logging
```
[the remark this answers is not in the document]
Comment (nkovela1) [reply keras_core/models/model.py]: Got it, fixed.

File: keras_core/models/model.py [hunk 2]
```diff
@@ -301,22 +304,32 @@ def save(self, filepath, overwrite=True, save_format="keras"):
...
             )
-        if not str(filepath).endswith(".keras"):
+        if not str(filepath).endswith((".keras", ".h5", ".hdf5")):
             raise ValueError(
-                "The filename must end in `.keras`. "
+                "The filename must end in `.keras` or `.h5`. "
                 f"Received: filepath={filepath}"
             )
+        if save_format:
+            if save_format in ["tf"]:
```
[the remark this answers is not in the document]
Comment (nkovela1) [reply keras_core/models/model.py]: Fixed, thanks

File: keras_core/models/model.py [hunk 3]
```diff
@@ -301,22 +304,32 @@ def save(self, filepath, overwrite=True, save_format="keras"):
...
+                raise ValueError(
+                    "The`'tf'` saved_model format are no longer supported "
+                    "via the `save_format` option. Please use the new "
+                    "`'keras'` format or the legacy `'h5'` format."
+                    f"Received: save_format={save_format}"
+                )
+            if save_format not in ["keras", "keras_v3", "h5"]:
```
[the remark this answers is not in the document]
Comment (nkovela1) [reply keras_core/models/model.py]: Rearranged tests into new class and added extra for code coverage.

**Review by fchollet — APPROVED** [review]
LGTM, thanks!

**Pull request merged**